### ISSUE TITLE
Session Expired Errors when buying Tickets

### ISSUE THREAD

**Issue by princekwame** [alfio-event/alf.io#903]
Hi There,

I've installed the docker version of this solution and I'm coming across a number of weird issues that I am hoping that someone here can help me with.

1. Session Expired.I've been getting this error pop-up during the Event ticketing process.

I create an event and set all the required configurations like start time of the event and timezone and create ticket categories, making sure that the the tickets times are within the event. I save and publish the event.

Once the event is published and I access it using the public URL to buy tickets, on this page I select a category and number of tickets I want and as soon I click continue, I get a "Session Expired" error 

Here are some images to show what I am seeing.

Ticket Selection page: Notice the number of tickets left.
![Screen Shot 2020-05-15 at 12 45 16 pm](https://user-images.githubusercontent.com/17686117/82007567-e7c24a80-96ad-11ea-864d-bfdb56916f48.png)

Once I click continue, this is what comes up..Session Expired.
![Screen Shot 2020-05-15 at 12 45 28 pm](https://user-images.githubusercontent.com/17686117/82007617-00cafb80-96ae-11ea-8a0b-716aa7da6a78.png)

When I click  "Back to main page", it takes me back to the event page again but this time, the number of available tickets have been reduced

![Screen Shot 2020-05-15 at 12 45 35 pm](https://user-images.githubusercontent.com/17686117/82007634-0d4f5400-96ae-11ea-9caa-3589d1c7d886.png)

and in fact if I cheche admin dashboard > Reservations, I can see that there is 1 pending reservation. One thing that's odd is expiry date of the "In progress" ticket, it seems that by default it's setting it an immediate time after the purchase.

![Screen Shot 2020-05-15 at 12 59 50 pm](https://user-images.githubusercontent.com/17686117/82007652-193b1600-96ae-11ea-8bd8-4686802e1b28.png)

Any help will be appreciated.

**Comment by princekwame** [comment]
Hey guys @syjer @salmar, I'm really stumped re this issue any thoughts?

**Comment by syjer** [comment]
hi @princekwame , have you checked if by accident you have set to 0 the following configuration:

(At system, organization or event level):

![Screenshot from 2020-05-16 14-56-55](https://user-images.githubusercontent.com/498146/82120345-8e2d4f00-9785-11ea-8b34-124de45d2bb3.png)

**Comment by princekwame** [comment]
hi @syjer 

Thanks for your response, I have not configured this setting either system or event level

![Screen Shot 2020-05-16 at 11 10 27 pm](https://user-images.githubusercontent.com/17686117/82120595-a70c2380-97ca-11ea-95d2-05e0cdde12de.png)
![Screen Shot 2020-05-16 at 11 10 01 pm](https://user-images.githubusercontent.com/17686117/82120597-a96e7d80-97ca-11ea-8d72-54dd5d8de4c1.png)

Just for your info as well this docker container is running behind a traefik proxy which is terminating HTTP/S and forwarding traffic onto container on port 8080, the traefik server has sticky sessions enabled.

**Comment by princekwame** [comment]
@syjer just an FYI, after posting my last comment I just thought to configure a time in there just to see what the behaviour is and it has actually fixed the session expired problem.

![Screen Shot 2020-05-16 at 11 24 32 pm](https://user-images.githubusercontent.com/17686117/82120904-9492e980-97cc-11ea-9dcf-4690686e0b82.png)

Which is great news, however, the timer seems to be off. I put 120 minutes as above but the time during the ticket purchase is only 1hr 8mins. Somehow 52mins is being removed...

I'm going to test further but this may be a bug. I'll get back to you with the results of my tests.

**Comment by princekwame** [comment]
@syjer here are a further update from my testing

1. Set timer to 1, default is 25; result still session expired
2. Set timer to 5, default is 25; result still session expired
3. Set timer to 15, default is 25; result still session expired
4. Set timer to 25, default is 25; result still session expired
4. Set timer to 30, default is 25; result still session expired
5. Set timer to 50, default is 25; there's no session expired message but it only gives me 9 secs to complete purchase
 
![Screen Shot 2020-05-16 at 11 46 23 pm](https://user-images.githubusercontent.com/17686117/82121334-927e5a00-97cf-11ea-98d4-5b4b78e02766.png)

6. Set timer to 60, default is 25; there's no session expired message but it only gives me 10mins to complete purchase.

![Screen Shot 2020-05-16 at 11 45 42 pm](https://user-images.githubusercontent.com/17686117/82121354-a88c1a80-97cf-11ea-9a6c-71a5697488d7.png)

**Comment by syjer** [comment]
hi @princekwame , have you checked at organization level too?

Maybe there is some issue with the timezone too?

**Comment by syjer** [comment]
@princekwame , looking at the code, I think your server has some issue with the time. 

The logic for setting the validity is a quite simple: "system date + X minutes"

**Comment by princekwame** [comment]
@syjer thanks for the info and yes, I compared the docker time/timezone with the docker host, there were a few differences,

1. The timezones were different between the docker host and container
2. Docker host was not syncing properly with NTP and the time was wrong even in the timezone it was using..quite a bit of time drift

I changed the docker host to UTC and fixed up the ntp sync, the container time reflected the docker host once it was working properly. I went back into Alf, removed the manual time I configured earlier completely, saved and tested the purchasing process and it works like magic using the default 25mins.

Thank you very much for your help in pointing me in the right direction!
